integration-tests/go-sql-server-driver/sqlserver_lock_test.go

```diff
@@ -0,0 +1,368 @@
+			t.Run("sql can run in root", func(t *testing.T) {
+				Setup(t)
+				cmd := rs.DoltCmd("sql", "-q", "show databases")
+				output, err := cmd.CombinedOutput()
+				outstr := string(output)
+				require.NoError(t, err)
+				require.Regexp(t, "db_one", outstr)
+				require.Regexp(t, "db_two", outstr)
+			})
+			t.Run("sql can run in db_one", func(t *testing.T) {
+				Setup(t)
+				cmd := dbOne.DoltCmd("sql", "-q", "show databases")
+				output, err := cmd.CombinedOutput()
+				outstr := string(output)
+				require.NoError(t, err)
+				require.Regexp(t, "db_one", outstr)
+			})
+		})
+		t.Run("With malformed sql-server.lock file in root", func(t *testing.T) {
+			Setup := func(t *testing.T) {
+				path := filepath.Join(rs.Dir, ".dolt/sql-server.lock")
+				err := os.WriteFile(path, []byte("1:3306:this_is_not_a_real_secret:extra:fields:make:it:fail"), 0600)
+				require.NoError(t, err)
+				t.Cleanup(func() { os.Remove(path) })
+			}
+			t.Run("sql-server can run in root", func(t *testing.T) {
+				Setup(t)
+				RunServerUntilEndOfTest(t, rs, &driver.Server{})
+			})
+			t.Run("sql-server can run in db_one", func(t *testing.T) {
+				Setup(t)
+				RunServerUntilEndOfTest(t, rs, &driver.Server{})
+			})
+			t.Run("sql can run in root", func(t *testing.T) {
+				Setup(t)
+				cmd := rs.DoltCmd("sql", "-q", "show databases")
+				output, err := cmd.CombinedOutput()
+				outstr := string(output)
+				require.NoError(t, err)
+				require.Regexp(t, "db_one", outstr)
+				require.Regexp(t, "db_two", outstr)
+			})
+			t.Run("sql can run in db_one", func(t *testing.T) {
+				Setup(t)
+				cmd := dbOne.DoltCmd("sql", "-q", "show databases")
+				output, err := cmd.CombinedOutput()
+				outstr := string(output)
+				require.NoError(t, err)
+				require.Regexp(t, "db_one", outstr)
+			})
+			t.Run("with dolt sql running in db_one", func(t *testing.T) {
+				RunDoltSQLUntilEndOfTest(t, dbOne)
+
+				t.Run("dolt sql -q fails in db_one", func(t *testing.T) {
+					Setup(t)
+					cmd := dbOne.DoltCmd("sql", "-q", "show databases")
+					_, err = cmd.CombinedOutput()
+					require.Error(t, err)
+				})
+				t.Run("dolt sql -q succeeds in root", func(t *testing.T) {
+					Setup(t)
+					cmd := rs.DoltCmd("sql", "-q", "show databases")
+					output, err := cmd.CombinedOutput()
+					require.NoError(t, err)
+					outstr := string(output)
+					require.Regexp(t, "db_one", outstr)
+					require.Regexp(t, "db_two", outstr)
+				})
+			})
+		})
+	})
+	t.Run("With Empty RepoStore", func(t *testing.T) {
+		u, err := driver.NewDoltUser()
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			u.Cleanup()
+		})
+
+		rs, err := u.MakeRepoStore()
+		require.NoError(t, err)
+
+		// TODO: This is strange behavior which the current implementation allows.
+		// If the top-level directory is empty when both servers are
+		// started, they can both run against it. Only one of their
+		// credential files will win the write.
+		t.Run("Can Run Two Servers At Once", func(t *testing.T) {
```

Comment:
This test was just intended to encode the behavior. Ideally it would start failing when we fix the behavior and then we could change the test to assert the new behavior (or the opposite order when someone goes to make that change...). Could also remove it of course...it's not verifying behavior we actually want, but it is verifying behavior we know exists today and that we would look like to see change in the future.